jujugui/static/gui/src/app/components/deployment-flow/payment/payment.js

```diff
@@ -124,39 +347,73 @@ YUI.add('deployment-payment', function() {
               Name and address
             </h2>
             {this._generateBusinessNameField()}
-            {this._generateAddressFields()}
+            <juju.components.GenericInput
+              disabled={disabled}
+              label="Email address"
+              ref="emailAddress"
+              required={true}
+              validate={[required]} />
+            {this._generateAddressFields('user')}
             <h2 className="deployment-payment__title">
               Payment information
             </h2>
             <juju.components.GenericInput
               disabled={disabled}
               label="Card number"
-              required={true} />
-            <div className="twelve-col">
-              <div className="six-col">
+              onChange={this._formatCardNumber}
+              ref="cardNumber"
+              required={true}
+              validate={[required, {
+                regex: /^[a-zA-Z0-9_-\s]{16,}/,
+                error: 'The card number is too short.'
+              }, {
+                regex: /^[a-zA-Z0-9_-\s]{0,23}$/,
+                error: 'The card number is too long.'
+              }, {
+                regex: /^[0-9\s]+$/,
+                error: 'The card number can only contain numbers.'
```

Comment:
The regex allows spaces, but shows an error if you put anything other than a space or a number in the field. The spaces are then removed here: https://github.com/juju/juju-gui/pull/2700/files#diff-d0c348641f90f06ae89fac8cb3f26069R198